ui/button.js

```diff
@@ -56,355 +39,267 @@ $.widget( "ui.button", {
 	defaultElement: "<button>",
 	options: {
 		disabled: null,
-		text: true,
+		showLabel: true,
 		label: null,
-		icons: {
-			primary: null,
-			secondary: null
+		icon: null,
+		iconPosition: "beginning",
+		classes: {
+			"ui-button": "ui-corner-all"
 		}
 	},
-	_create: function() {
-		this.element.closest( "form" )
-			.unbind( "reset" + this.eventNamespace )
-			.bind( "reset" + this.eventNamespace, formResetHandler );
 
-		if ( typeof this.options.disabled !== "boolean" ) {
-			this.options.disabled = !!this.element.prop( "disabled" );
-		} else {
-			this.element.prop( "disabled", this.options.disabled );
-		}
+	_getCreateOptions: function() {
+		var disabled,
+			options =  this._super() || {};
 
-		this._determineButtonType();
-		this.hasTitle = !!this.buttonElement.attr( "title" );
+		this.isInput = this.element.is( "input" );
+		this.originalLabel = this.isInput ? this.element.val() : this.element.html();
```

Comment:
This line should be grouped with the `if` below. Either moves this down or move the `if` up.